README.md

```diff
@@ -543,3 +517,233 @@ Webhooks for prow should already be configured according to these [instructions]
 need to set hooks per repository.
     * Use https://prow.k8s.io/hook as the target
     * Get HMAC token from k8s test team
+
+## Guidelines For Writing An Argo Workflow For An E2E Test
+
+This section provides guidelines for writing Argo workflows to use as E2E tests
+
+This guide is complementary to the [E2E testing guide for TFJob operator](https://github.com/kubeflow/tf-operator/blob/master/e2e_testing.md)
+which describes how to author tests to performed as individual steps in the workflow.
+
+Some examples to look at
+
+  * code_search.jsonnet in kubeflow/examples
+
+## Adding an E2E test for a new repository
+
+We use prow to launch Argo workflows. Here are the steps to create a new E2E test for a repository. This assumes prow is already
+configured for the repository (see these [instructions](#prow-setup) for info on setting up prow).
+
+### Adding an E2E test to a repository
+
+Follow these steps to add a new test to a repository.
+
+1. Create a ksonnet App in that repository and define an Argo workflow if one doesn't already exist
+   * We use ksonnet apps defined in each repository to define the Argo workflows corresponding to E2E tests
+   * If a ksonnet app already exists you can just define a new component in that app
+
+     1. Create a .jsonnet file (e.g by copying an existing .jsonnet file)
+
+        * Change the import for the params to use the newly defined component
+
+     1. Update the `params.libsonnet` to add a stanza to define params for the new component
+
+   * You can look at `prow_config.yaml` (see below) to see which ksonnet apps are already defined in a repository.
+
+1. Modify the `prow_config.yaml` at the root of the repo to trigger your new test.
+
+   * If `prow_config.yaml` doesn't exist (e.g. the repository is new) copy one from an existing repository ([example](https://github.com/kubeflow/kubeflow/blob/master/prow_config.yaml)).
+     
+   * `prow_config.yaml` contains an array of workflows where each workflow defines an E2E test to run; example
+
+       ```
+       workflows:
+        - app_dir: kubeflow/testing/workflows
+          component: workflows
+          name: unittests
+          job_types:
+            - presubmit
+          include_dirs:
+            - foo/*
+            - bar/*
+              params:
+          params:
+            platform: gke
+            gkeApiVersion: v1beta1
+       ```
+
+       * **app_dir**: Is the path to the ksonnet directory within the repository. This should be of the form `${GITHUB_ORG}/${GITHUB_REPO_NAME}/${PATH_WITHIN_REPO_TO_KS_APP}`
+       * **component**: This is the name of the ksonnet component to use for the Argo workflow
+       * **name**: This is the base name to use for the submitted Argo workflow. A unique suffix is added to give each run a unique name.
```

Comment:
I think there are some limitations to the length of the name, might be worth mentioning here.

Reply:
Done. I suggest checking the length in the ksonnet file and adjusting it as needed. In the section on the Argo spec I explicitly call out the 63 character limit on Argo workflow names.